spec.html

```diff
@@ -12,194 +12,174 @@ <h1>Locale Objects</h1>
-        1. Return *undefined*.
-      </emu-alg>
-    </emu-clause>
-
-    <emu-clause id="sec-unicodeextensionvalue" aoid="UnicodeExtensionValue">
-      <h1>UnicodeExtensionValue ( _extension_, _key_ )</h1>
-
-      <p>The abstract operation UnicodeExtensionValue is called with _extension_, which must be a Unicode locale extension sequence, and String _key_. This operation returns the type subtags for _key_ by performing the following steps:</p>
-
-      <emu-alg>
-        1. Let _range_ be FindExtension(_extension_, _key_).
-        1. If _range_ is *undefined*, return *undefined*.
-        1. If _range_.[[Start]] + 4 &ge; _range_.[[End]], return an empty String.
-        1. Otherwise, return a String value equal to the substring of _extension_ from _range_.[[Start]] + 4 (inclusive) to _range_.[[End]] (exclusive).
-      </emu-alg>
-    </emu-clause>
-
-    <emu-clause id="sec-unused-extensions" aoid="UnusedExtensions">
-      <h1>UnusedExtensions ( r )</h1>
-      <emu-alg>
-        1. Let _extension_ be _r_.[[extension]].
-        1. For _key_ in %Locale%.[[RelevantExtensionKeys]],
-          1. Let _done_ be *false*.
-          1. Repeat, while _done_ is *false*,
-            1. Let _range_ be FindExtension(_extension_, _key_).
-            1. If _range_ is *undefined*, set _done_ to *true*.
-            1. Otherwise, set _extension_ to the concatenation of the substring of _extension_ from the beginning to _range_.[[Start]] with the substring of _extension_ from _range_.[[End]] to the end of the string.
-        1. If _extension_ begins with `"-u-"`, set _extension_ to the substring of _extension_ beginning from its fourth index.
-        1. Return _extension_.
+        1. Let _key_ be `"attributes"`.
+        1. Let _value_ be the empty String.
+        1. Let _k_ be 3.
+        1. Repeat, while _k_ &lt; _size_
+          1. Let _e_ be ! Call(%StringProto_indexOf%, _extension_, « `"-"`, _k_ »).
+          1. If _e_ = -1, let _len_ be _size_ - _k_; else let _len_ be _e_ - _k_.
+          1. Let _subtag_ the String value equal to the substring of _extension_ consisting of the code units at indices _k_ (inclusive) through _k_ + _len_ (exclusive).
+          1. If _len_ = 2, then
+            1. If _keywords_ does not contain an element whose [[Key]] is the same as _key_, then
+              1. Append the Record{[[Key]]: _key_, [[Value]]: _value_} to _keywords_.
+            1. Let _key_ be _subtag_.
+            1. Let _value_ be the empty String.
+          1. Else,
+            1. If _value_ is not the empty String, then
+              1. Let _value_ be the string-concatenation of _value_ and `"-"`.
+            1. Let _value_ be the string-concatenation of _value_ and _subtag_.
+          1. Let _k_ be _k_ + _len_ + 1.
+        1. If _keywords_ does not contain an element whose [[Key]] is the same as _key_, then
+          1. Append the Record{[[Key]]: _key_, [[Value]]: _value_} to _keywords_.
```

Comment:
Should the previous two lines be indented one level further?

Reply:
No, it's at the correct level, the last element is added to `keywords` after the loop. (Elements are added to `keywords` when a new Unicode extension key was found or the end of the Unicode extension sequence has been reached.)

Reply:
Oh, I see.